hypertrace-graphql-labels-schema-impl/src/main/java/org/hypertrace/graphql/label/deserialization/LabelApplicationRuleDataDeserializationConfig.java

```diff
@@ -0,0 +1,152 @@
+package org.hypertrace.graphql.label.deserialization;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.Module;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import java.util.List;
+import lombok.NoArgsConstructor;
+import lombok.Value;
+import lombok.experimental.Accessors;
+import org.hypertrace.core.graphql.deserialization.ArgumentDeserializationConfig;
+import org.hypertrace.graphql.label.schema.rule.Action;
+import org.hypertrace.graphql.label.schema.rule.Condition;
+import org.hypertrace.graphql.label.schema.rule.LabelApplicationRuleData;
+import org.hypertrace.graphql.label.schema.rule.LeafCondition;
+import org.hypertrace.graphql.label.schema.rule.StaticLabels;
+import org.hypertrace.graphql.label.schema.rule.StringCondition;
+import org.hypertrace.graphql.label.schema.rule.UnaryCondition;
+import org.hypertrace.graphql.label.schema.rule.ValueCondition;
+
+public class LabelApplicationRuleDataDeserializationConfig
+    implements ArgumentDeserializationConfig {
+  @Override
+  public String getArgumentKey() {
+    return LabelApplicationRuleData.ARGUMENT_NAME;
+  }
+
+  @Override
+  public Class<?> getArgumentSchema() {
+    return LabelApplicationRuleData.class;
+  }
+
+  @Override
+  public List<Module> jacksonModules() {
+    return List.of(
+        new SimpleModule()
+            .addAbstractTypeMapping(
```

Comment:
Sloppy review on my part - we can't get rid of the whole file since the arg name/schema mapping is still needed above, but all the jackson mappings are unnecessary if they're already declared elsewhere.